test/integration/listener_extension_discovery_integration_test.cc

```diff
@@ -0,0 +1,317 @@
+#include "envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.pb.h"
+#include "envoy/service/discovery/v3/discovery.pb.h"
+#include "envoy/service/extension/v3/config_discovery.pb.h"
+
+#include "test/common/grpc/grpc_client_integration.h"
+#include "test/integration/filters/test_listener_filter.h"
+#include "test/integration/filters/test_listener_filter.pb.h"
+#include "test/integration/integration.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+class ListenerExtensionDiscoveryIntegrationTest : public Grpc::GrpcClientIntegrationParamTest,
+                                                  public BaseIntegrationTest {
+public:
+  ListenerExtensionDiscoveryIntegrationTest()
+      : BaseIntegrationTest(ipVersion(), ConfigHelper::baseConfig() + R"EOF(
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.tcp_proxy
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
+          stat_prefix: tcp_stats
+          cluster: cluster_0
+)EOF"),
```

Comment:
consider putting this + ConfigHelper::baseConfig in a function in the anonymous namespace so this part becomes more readable

Reply:
ACK